ext/datadog_profiling_native_extension/stack_recorder.c

```diff
@@ -330,6 +327,8 @@ static VALUE _native_new(VALUE klass) {
   // Note: Any exceptions raised from this note until the TypedData_Wrap_Struct call will lead to the state memory
   // being leaked.
 
+  state->heap_clean_after_gc_enabled = true;
```

Comment:
Given the current status of this is default false, why are we setting to `true` here? I know it'll get overriden by `_native_initiatlize` but seems unnecessary?

Reply:
Yeah, you're right, in hindsight this may be a bit misleading. Changed in 6494eee31afbe68e9008804dde825bc3e79556c6 .